pretyping/reductionops.ml

```diff
@@ -89,11 +104,19 @@ module ReductionBehaviour = struct
     | UnfoldWhen x -> UnfoldWhen (more_args_when k x)
     | UnfoldWhenNoMatch x -> UnfoldWhenNoMatch (more_args_when k x)
 
+(* We need to have a fast way to know the set of all constants that
+  have the NeverUnfold flag.  Therefore, the table has a distinct subpart
+  that is this set. *)
   let table =
-    Summary.ref (GlobRef.Map.empty : t GlobRef.Map.t) ~name:"reductionbehaviour"
+    Summary.ref ((GlobRef.Set.empty, GlobRef.Map.empty) :
+                   GlobRef.Set.t * t GlobRef.Map.t) ~name:"reductionbehaviour"
 
   let load _ (_,(_,(r, b))) =
-    table := GlobRef.Map.add r b !table
+    table := ((match b with
+                NeverUnfold ->
```

Comment:
Note that this is wrong: when you add a ref to one side, you need to remove it from the other. Otherwise the semantics of maps is not respected.